vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java

```diff
@@ -68,7 +88,47 @@ public void test_deleteImagetBlockById_whenIdIsNull() throws BlockDoesNotExistEx
         String imageBlockId = null;
         managerToTest.deleteImageBlockById(null);
         Mockito.verify(imageBlockRepo, Mockito.never()).deleteById(imageBlockId);
+    }
+    
+    @Test
+    public void test_deleteBiblioBlockById_success() throws BlockDoesNotExistException, ReferenceListDeletionForBiblioException {
+        String biblioBlockId = "CON000000002";
+        managerToTest.deleteBiblioBlockById(biblioBlockId);
+        Mockito.verify(biblioBlockRepo).deleteById(biblioBlockId);
+    }
+    
+    @Test(expected = BlockDoesNotExistException.class)
+    public void test_deleteBiblioBlockById_forNonExistentId() throws BlockDoesNotExistException, ReferenceListDeletionForBiblioException {
+        String biblioBlockId = "notARealId";
+        Mockito.doThrow(EmptyResultDataAccessException.class).when(biblioBlockRepo).deleteById(biblioBlockId);
+        managerToTest.deleteBiblioBlockById(biblioBlockId);
+    }
+
+    @Test
+    public void test_deleteBiblioBlockById_whenIdIsNull() throws BlockDoesNotExistException, ReferenceListDeletionForBiblioException {
+        String biblioBlockId = null;
+        managerToTest.deleteBiblioBlockById(null);
+        Mockito.verify(biblioBlockRepo, Mockito.never()).deleteById(biblioBlockId);
+    }
+    
+    @Test
+    public void test_createBiblioBlock_success() {
+
+        Integer contentOrder = 1;
 
+        Slide slide = new Slide();
+        slide.setId("slide1");
+        
+        IBiblioBlock biblioBlock = new BiblioBlock();
+        biblioBlock.setBiblioTitle("TestTitle");
+        biblioBlock.setDescription("Test Description");
+        
+        biblioBlock.setContentOrder(contentOrder);
+        biblioBlock.setSlide(slide);
+        
+        when(slideManager.getSlide(slide.getId())).thenReturn(slide);
+        managerToTest.createBiblioBlock(slide.getId(), biblioBlock);
```

Comment:
doesn't this method return something? your test should check it returns the correct things.

Reply:
I think here in every block we are making a db call and Mockito is not actually saving anything. It's a mock and that's why we are checking whether save method is called. We are doing this for all the blocks i.e image block, text block in the same way. Please let me know if this isn't the correct way.

Reply:
`createBiblioBlock` returns the block with an id and sets the slide. these are two things you can test for and are better suited for testing, since that's the result you want to see (even if a different framework/method would be used that dosen't call save, the test should still pass).

Reply:
```
@Override
    public IBiblioBlock createBiblioBlock(String slideId, IBiblioBlock biblio) {
        ISlide slide = slideManager.getSlide(slideId);
        biblio.setSlide(slide);
        return biblioBlockRepo.save((BiblioBlock) biblio);
    }
```
This method returns a biblioblock. You can set the mock up in a way that it sets an id when saved. I believe @MaulikLim or @AvirupBiswas have done this before.

Reply:
Ok Thank you. I addressed this now